packages/shared/components/molecules/SelectorInput.svelte

```diff
@@ -1,16 +1,24 @@
 <script lang="typescript">
     import { localize } from '@core/i18n'
     import { Modal, TextInput } from 'shared/components'
+    import { fade } from 'svelte/transition'
 
     export let value: string = ''
     export let error: string = ''
     export let disabled: boolean = false
     export let labelLocale: string = ''
     export let modal: Modal = undefined
     export let inputElement: HTMLInputElement = undefined
+    export let options = []
+    export let onClick: (val: unknown) => void = undefined
 
     let hasFocus: boolean
 
+    function handleClick(option: any) {
+        modal?.close()
+        onClick ?? onClick(option)
```

Comment:
Why not call `onClick(option)` directly?

Reply:
because it could be undefined. 
i could use `onClick?.(option)`